endpoints/openrtb2/amp_auction.go

```diff
@@ -219,6 +220,7 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		ao.Errors = append(ao.Errors, err)
 		return
 	}
+	labels.DebugEnabled = debugLog.DebugFlag
```

Comment:
What is the purpose of this line? ... since it occurs after the `HoldAuction` call. Wouldn't the metrics for an incoming request have been recorded before hitting this line? Same applies to `openrtb/auction.go`

Reply:
If I understand this correctly, metrics will be written at the end of endpoint execution. It's defined in defer section, L122:
```
defer func() {
		deps.metricsEngine.RecordRequest(labels)
		deps.metricsEngine.RecordRequestTime(labels, time.Since(start))
		deps.analytics.LogAmpObject(&ao)
	}()
```
Auction, video and amp endpoints have this approach.

Reply:
Yeah exactly like Veronika said above! And when I sent test postman requests, it acted like this as well!

Reply:
Ah. Got it. This code is a bit too clever IMHO, but I agree it's beyond the scope of this PR to fix. Can you add a comment before these lines to make it clear what's going on to my future self?